### PR TITLE
update vsphere arguments to reflect changes in 0.4.2 provider

### PR DESCRIPTION
**Pull request by jonmosco**
Change datacenter to datacenter_id. Updated provider.tf and variables.tf to reflect changes.

Default variable to be "vm" since this will primarily be used to create VMs.

See https://github.com/terraform-providers/terraform-provider-vsphere/blob/master/CHANGELOG.md

Breaking changes presented in 0.4.0 to vsphere_folder that require updating.

### CODE REVIEW

File: platforms/vmware/variables.tf [hunk 1]
```diff
@@ -25,6 +25,12 @@ variable "tectonic_vmware_folder" {
   description = "vSphere Folder to create and add the Tectonic nodes"
 }
 
+variable "tectonic_vmware_type" {
+  type        = "string"
+  description = "The type of folder to create. Allowed options: datacenter,host,vm, datastore, and network."
```
Comment (mxinden) [inline platforms/vmware/variables.tf]: `s/datacenter,host,vm,/datacenter, host, vm,/`

**Comment by mxinden** [comment]
ok to test

**Comment by jonmosco** [comment]
Spacing fixed.

**Comment by squat** [comment]
@jonmosco thanks for fixing that. The basic tests are failing currently because we need to update the examples. Please run `make docs examples`, commit the result like `git commit -m "docs,examples: regenerate` and push, then we should be good.

**Comment by jonmosco** [comment]
Done.  I did not rebase this, let me know if that is necessary.

**Comment by squat** [comment]
Can you push the changes? I am not seeing anything new on github

**Comment by jonmosco** [comment]
Done.

**Comment by squat** [comment]
ok to test

**Comment by squat** [comment]
ok to test

**Comment by squat** [comment]
ok to test

**Comment by squat** [comment]
ok to test

**Comment by squat** [comment]
@jonmosco CI is complaining that the examples have not been updated. Looking at the changes, I agree. Did you `git add examples` to commit the new examples as well?

**Comment by jonmosco** [comment]
Examples were not generated:  Makefile:101: *** "terraform-examples
revision >= 83d7ad6 is required (
https://github.com/s-urbaniak/terraform-examples)".  Stop.

**Comment by squat** [comment]
ok this explains the issue then. You will need to install that binary then via `go get -u github.com/s-urbaniak/terraform-examples`. Please let me know if you'd like me to add that change to this PR or if you prefer to do it yourself.

**Comment by jonmosco** [comment]
Done, and also squashed.

**Pull request merged**